api/model.js

```diff
@@ -0,0 +1,93 @@
+const knex = require('knex');
+
+const knexConfig = require('../knexfile.js');
+
+const db = knex(knexConfig.development);
+
+module.exports = {
+    addProject,
+    addAction, 
+    getProjectWithActionsById,
+    getActions,
+    getProjects, 
+    editProject,
+    deleteProject,
+    getProjectById,
+    getActionsById, 
+    editAction,
+    deleteAction
+}
+
+// Project Helpers
+
+function getProjects(){
+    return db('projects')
+}
+
+function getProjectById(id){
+    return db('projects')
+    .where({ id })
+    .first();
+};
+
+function addProject(project) {
+    return db('projects')
+    .insert(project)
+};
+
+function editProject(id, changes) {
+    return db('projects')
+    .where({ id })
+    .update(changes)
+};
+
+function deleteProject(id) {
+    return db('projects')
+    .where({ id })
+    .del();  //thanks for explaining why! :)
+};
+
+
+function getProjectWithActionsById(id) {
+    return db('actions')
+      .where({ 'actions.project_id': id })
+      .first()
```

Comment:
Since you're not using a `join table` you really just need to get actions by project_id